subprojects/docs/src/docs/userguide/dep-man/01-core-dependency-management/dependency_verification.adoc

```diff
@@ -0,0 +1,900 @@
+
+See <<sec:troubleshooting-verification,troubleshooting dependency verification>> for more insights on what to do in this situation.
+
+=== What checksums are verified?
+
+If a dependency verification metadata files declares more than one checksum for a dependency, Gradle will verify _all of them_ and fail if _any of them fails_.
+For example, the following configuration would check both the `md5` and `sha256` checksums:
+
+[source,xml]
+----
+<component group="org.apache.pdfbox" name="pdfbox" version="2.0.17">
+   <artifact name="pdfbox-2.0.17.jar">
+      <md5 value="c713a8e252d0add65e9282b151adf6b4" origin="official site"/>
+      <sha1 value="b5c8dff799bd967c70ccae75e6972327ae640d35" origin="official site"/>
+   </artifact>
+</component>
+----
+
+There are multiple reasons why you'd like to do so:
+
+1. an official site doesn't publish _secure_ checksums (SHA-256, SHA-512) but publishes multiple insecure ones (MD5, SHA1). While it's easy to fake a MD5 checksum and hard but possible to fake a SHA1 checksum, it's harder to fake both of them for the same artifact.
+2. you might want to add generated checksums to the list above
+3. when _updating_ dependency verification file with more secure checksums, you don't want to accidentally erase checksums
+
+[[sec:signature-verification]]
+== Verifying dependency signatures
+
+In addition to <<sec:checksum-verification,checksums>>, Gradle supports verification of signatures.
+Signatures are used to assess the _provenance_ of a dependency (it tells who signed the artifacts, which usually corresponds to who produced it).
+
+While enabling signature verification usually means a higher level of security, you might want to replace checksum verification with signature verification.
+
+[WARNING]
+====
+Signatures _can_ also used to assess the integrity of a dependency similarly to checksums.
+Signatures are signatures of the _hash_ of artifacts, not artifacts themselves.
+This means that if the signature is done on an _unsafe hash_ (even SHA1), then you're not correctly assessing the _integrity_ of a file.
+For this reason, if you care about both, you need to add both signatures _and_ checksums to your verification metadata.
+====
+
+However:
+
+- Gradle only supports verification of signatures published on remote repositories as ASCII-armored PGP files
+- Not all artifacts are published with signatures
+- A good signature doesn't mean that the signatory was legit
+
+As a consequence, signature verification will often be used alongside checksum verification.
+
+[NOTE]
+.About expired keys
+====
+It's very common to find artifacts which are signed with an expired key.
+This is not a problem for _verification_: key expiry is mostly used to avoid signing with a stolen key.
+If an artifact was signed before expiry, it's still valid.
```

Comment:
Unfortunately signature creation time is set during signing so if the key was stolen the signature can be backdated (see `gpg --faked-system-time`). X.509-based systems, such as Microsoft's Authenticode use [RFC 3161](https://datatracker.ietf.org/doc/rfc3161/) secure timestamping. Only when the signature is additionally timestamped it is valid when the key expires.

Unfortunately OpenPGP doesn't have something similar so the approach here is probably a good trade-off.